### PR TITLE
PageCache storing original header names

### PR DESCRIPTION
**Pull request by bizley**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ~
| Breaks BC?    | ✔️ a bit
| Fixed issues  | #19187 

I've added additional data to HeaderCollection storing the original name of the header so it can be provided with `toArray(true)`. This change makes a small BC with the filtering option for PageCache since now the names are compared to the case-sensitive versions of the headers but I can add some code to keep it as before.

### CODE REVIEW

File: framework/UPGRADE.md [hunk 1]
```diff
@@ -51,6 +51,12 @@ if you want to upgrade from version A to version C and there is
 version B between A and C, you need to follow the instructions
 for both A and B.
 
+Upgrade from Yii 2.0.44
+-----------------------
+
+* `yii\filters\PageCache::$cacheHeaders` now takes a case-sensitive list of header names since PageCache is no longer 
+  storing the normalized versions of them.
```
Comment (samdark) [inline framework/UPGRADE.md:58]: Need to mention about what should be done in user apps.

**Review by machour — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

**Comment by samdark** [comment]
👍

File: framework/web/HeaderCollection.php [hunk 2]
```diff
@@ -163,16 +170,25 @@ public function remove($name)
...
 
     /**
      * Returns the collection as a PHP array.
      * @return array the array representation of the collection.
      * The array keys are header names, and the array values are the corresponding header values.
+     * Since 2.0.45 you can pass true here to get the headers list of original header names (case-sensitive) instead of
+     * default normalized (case-insensitive) ones.
      */
-    public function toArray()
+    public function toArray($originalNames = false)
```
Comment (rob006) [inline framework/web/HeaderCollection.php:183]: This change breaks BC - if someone overwrites this method he will get fatal error after this change.

Comment (samdark) [reply framework/web/HeaderCollection.php:183]: @bizley would you please add it to UPGRADE?

Comment (rob006) [reply framework/web/HeaderCollection.php:183]: Why we're introducing such BC break in the first place? We could add another method.

Comment (schmunk42) [reply framework/web/HeaderCollection.php:183]: @samdark I agree with @rob006 - it's a method signature change for a public method, we should not do that.

Comment (bizley) [reply framework/web/HeaderCollection.php:183]: Yes, right, why haven't I think of that. I'll add a new one.

**Pull request merged**